src/compiler/checker.ts

```diff
@@ -3530,6 +3531,24 @@ namespace ts {
             return exports;
         }
 
+        function forEachExportAndPropertyOfModule(moduleSymbol: Symbol, cb: (symbol: Symbol, key: __String) => void): void {
+            const exports = getExportsOfModule(moduleSymbol);
+            exports.forEach((symbol, key) => {
+                if (!isReservedMemberName(key)) {
+                    cb(symbol, key);
+                }
+            });
+            const exportEquals = resolveExternalModuleSymbol(moduleSymbol);
+            if (exportEquals !== moduleSymbol) {
+                const type = getTypeOfSymbol(exportEquals);
+                if (shouldTreatPropertiesOfExternalModuleAsExports(type)) {
+                    getPropertiesOfType(type).forEach(symbol => {
+                        cb(symbol, symbol.escapedName);

```

Comment:
I would have had to refactor or duplicate a lot of checker code to get the actual map key of `symbol.members` here, so I’m _hoping_ that the map key for a _property_ is always `symbol.escapedName`, which is not always true for an _export_. I added a debug assertion here while running tests and it never triggered, but I don’t want to leave it in production code because this is an extremely hot path (can easily hit tens of thousands of times while building a large export info map).

Reply:
What's the penalty for providing the wrong key in this PR?